src/vm/eventpipe.cpp

```diff
@@ -587,30 +589,21 @@ void EventPipe::WriteEventInternal(
         BYTE *pData = payload.GetFlatData();
         if (pThread != nullptr && pRundownSession != nullptr && pData != nullptr)
         {
-            // Write synchronously to the file.
-            // We're under lock and blocking the disabling thread.
-            // This copy occurs here (rather than at file write) because
-            // A) The FastSerializer API would need to change if we waited
-            // B) It is unclear there is a benefit to multiple file write calls
-            //    as opposed a a buffer copy here
-            EventPipeEventInstance instance(
-                event,
-                GetCurrentProcessorNumber(),
-                pEventPipeThread->GetOSThreadId(),
-                pData,
-                payload.GetSize(),
-                pActivityId,
-                pRelatedActivityId);
-            instance.EnsureStack(*pRundownSession);
```

Comment:
Can you check if the `pStack` is a legitimate value? By removing the call to `instance.EnsureStack()` you are removing a whole stack walk here.

Reply:
https://github.com/dotnet/coreclr/blob/bf3dce315eb9fe8da4491b711fecddb4116e2314/src/vm/eventpipebuffermanager.cpp#L372-L376

https://github.com/dotnet/coreclr/blob/bf3dce315eb9fe8da4491b711fecddb4116e2314/src/vm/eventpipeeventinstance.cpp#L64-L70

This should behave the same as before, but the `EnsureStack` will happen without a function call inside `WriteEvent`.  It seems like if `pStack` was junk data before, it would still be junk data now.  We could explicitly set it `NULL`, though.

Reply:
This comment seems moot. It appears that `session.RundownEnabled()` is true, so both paths will not collect the stacks anyway.